contrib_bots/lib/gmail.py

```diff
@@ -0,0 +1,198 @@
+            messages with "@gmail".
+
+            Enter messages in the following format:
+            '@gmail <recipient_email> "<subject>" "<body>" --label <label>'
+            The --label tag is optional and can be left out.
+            '''
+
+    def help(self, client, message):
+        content = ("This plugin will allow users to send emails from a"
+                   "personal gmail account. Users should preface"
+                   "messages with '@gmail'."
+
+                   "Enter messages in the following format:"
+                   '@gmail <recipient_email> "<subject>" "<body>" --label <label>'
+                   "The --label tag is optional and can be left out. Note the quotation"
+                   " marks around the subject and body.")
+
+        client.send_message(dict(
+            type = 'private',
+            to = message['sender_email'],
+            sender = self.bot_email,
+            content = content
+        ))
+
+    def triage_message(self, message, client):
+        # return True if we want to (possibly) response to this message
+
+        original_content = message['content']
+        self.bot_email = client.email
+
+        # This next line of code is defensive
+        if message['display_recipient'] == 'gmail':
+            return False
+
+        if original_content.startswith('@gmail') and 'help' in original_content:
+            help(client, message)
+            return False
+
+        return original_content.startswith('@gmail')
+
+    def handle_message(self, message, client, state_handler):
+        original_content = message['content']
+        original_sender = message['sender_email']
+
+        try:
+            recipient, subject, body, label = self.parse_input(original_content)
+        except TypeError:
+            content = ("The input format used was incorrect. The correct format is: "
+                       '@gmail <recipient> "<subject>" "<body>"')
+
+            client.send_message(dict(
+                type = 'private',
+                to = original_sender,
+                sender = self.bot_email,
+                content = content
+            ))
+
+            return False
+
+        service = get_service()
+        email = create_message(body, subject, recipient)
+        result = send_message(service, email)
+
+        label_result = False
+        if label:
+            create_label(service, label)
+            label_id = get_label_id(service, label)
+            label_result = add_label(service, label_id, result['id'])
+
+        if 'SENT' in result['labelIds']:
+            content = 'Email sent to {}'.format(recipient)
+            if label_result:  # Enters this block if the label couldn't be added
+                              # or if there was no label
+                content += ' without label.'
+            else:
+                content += '.'
+        else:
+            content = 'Could not send email to {}. Please try again.'.format(recipient)
+
```

Comment:
Can all this code that is building up the content to send back to the user be put in a function?   Basically, everything between lines 97 and line 111.